src/perl/lib/wtsi_clarity/clarity/process.pm

```diff
@@ -85,10 +86,38 @@ has 'input_artifacts' => (
 sub _build_input_artifacts {
   my $self = shift;
 
+  my @input_states = @{$self->input_states()};
```

Comment:
Do you really need this variable? Why are you not passing directly `$self->input_states()` as batch_retrieve parameter's in line 91?